### PR TITLE
Extract commands from new protocol inside a dedicated library

### PR DESCRIPTION
**Pull request by xvw**
**This PR is based on `voodoos/502` (https://github.com/ocaml/merlin/pull/1757)**

In order to easily enable tunneling, via a custom query from `ocaml-lsp`, a large part of the argument processing logic (argv) can be used. This branch separates the command invocation logic (and the dispatching of CLI arguments to useful `merlin-lib` functionality) into a dedicated library, which can be used from within `ocaml-lsp`.

### CODE REVIEW

File: src/commands/dune [hunk 1]
```diff
@@ -0,0 +1,14 @@
+(library
+  (name merlin_commands)
+  (public_name merlin-lib-commands)
```
Comment (voodoos) [inline src/commands/dune:3]:
```
(public_name merlin-lib.commands)
```
If it does not actually introduce new dependencies, we might simply ship it as a part of `merlin-lib`

**Review by voodoos — COMMENTED** [review]
Thanks @xvw !

It's nice to realize that `Query_json` actually doesn't depends on Yojson :-)

Did you also check that other important parsing function such as the one parsing the flags to build the configuration are already exposed by merlin lib ?

File: src/commands/dune (hunk 1 above)
Comment (xvw) [reply src/commands/dune]: Done in the last commit https://github.com/ocaml/merlin/pull/1758/commits/24e97948e1d3f3774484f26e6c18338ac266d02a

**Review by voodoos — APPROVED** [review]
These changes LGTM.

Were you able to confirm that this is actually enough to implement the "tunneling" on the LSP side ?

**Comment by xvw** [comment]
> Thanks @xvw !
> 
> It's nice to realize that `Query_json` actually doesn't depends on Yojson :-)
> 
> Did you also check that other important parsing function such as the one parsing the flags to build the configuration are already exposed by merlin lib ?

Yes. I am integrating test into my `ocaml-lsp` branch. I'll do the PR today.

**Comment by xvw** [comment]
Done!

**Pull request merged**